### DISCUSSION TITLE
Piano-like keyboard layout?

### DISCUSSION THREAD

**Discussion by Lossdeys** [SynthstromAudible/DelugeFirmware#1165] [Ideas]
Greetings everyone!
Is it possible to make key layout that looks like piano:
row of black keys
row of white keys
and so on.
Sadly I'm struggling with memorizing isomorphic layout (and find new In-Key not much helpful too). Yep, hardware MIDI-keyboard helps a lot, but sometimes it's inconvenient to use (couch play, for example).

**Comment by sapphire-arches** [comment]
We have added the ability to adjust the row step -- if you hold shift and turn the horizontal encoder while in the keyboard view you can set it to 12 or 0 which will make the whole column the same note relative to the root. If that's not what you're after maybe we can add it too, though a doodle of some sort might help better understand what you're after.

**Comment by Lossdeys** [comment]
Hello, sapphire-arches. I suggest mimicking real piano keyboard on Deluge pad rows. Of course, this layout looks a bit clumsy, but maybe it's more familiar and friendly to every piano/midi-keyboard player.
![Keyboard layout](https://github.com/SynthstromAudible/DelugeFirmware/assets/158769084/cdcafecd-efe3-4e61-8d6b-46de9ff74442)
I don't know which option (1 or 2) is more comfortable though.
P.S. In-key layout is great and rather close to mode "white keys only" for some root notes, but still works only in scales, not whole keyboard.

**Reply by sapphire-arches** [reply]
I see, this should be possible to implement. I'm not sure anyone will take it on but the keyboard API is one thing we've made much more extensible so it would be a good project for a newcomer to take on!

**Comment by Radius9999** [comment]
I would love to see this implemented also.  As a keyboardist, training my brain to view notes differently is rather frustrating.
